src/cpp/src/rag/text_embedding_pipeline.cpp

```diff
@@ -0,0 +1,290 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "openvino/genai/rag/text_embedding_pipeline.hpp"
+
+#include "debug_utils.hpp"
+#include "openvino/genai/tokenizer.hpp"
+#include "openvino/opsets/opset.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "openvino/opsets/opset3.hpp"
+#include "openvino/opsets/opset8.hpp"
+#include "utils.hpp"
+
+namespace {
+using namespace ov::genai;
+
+ov::AnyMap remove_config_properties(const ov::AnyMap& properties) {
+    auto properties_copy = properties;
+
+    properties_copy.erase(max_length.name());
+    properties_copy.erase(pooling_type.name());
+    properties_copy.erase(normalize.name());
+    properties_copy.erase(embed_instruction.name());
+    properties_copy.erase(query_instruction.name());
+
+    return properties_copy;
+}
+}  // namespace
+
+namespace ov {
+namespace genai {
+using utils::read_anymap_param;
+
+TextEmbeddingPipeline::Config::Config(const ov::AnyMap& properties) {
+    read_anymap_param(properties, ov::genai::max_length.name(), max_length);
```

Comment:
```suggestion
    read_anymap_param(properties, max_length.name(), max_length);
```
as we are already within `ov::genai` namespace

Reply:
We are in `Config` struct method where we have `max_length` struct member. So we need to differentiate between `max_length` property and struct member.